os/drivers/audio/alc5658.c

```diff
@@ -1305,39 +1430,39 @@ FAR struct audio_lowerhalf_s *alc5658_initialize(FAR struct i2c_dev_s *i2c, FAR
 
 	/* Allocate a ALC5658 device structure */
 	priv = (FAR struct alc5658_dev_s *)kmm_zalloc(sizeof(struct alc5658_dev_s));
-	if (priv) {
-
-		/* Initialize the ALC5658 device structure.  Since we used kmm_zalloc,
-		 * only the non-zero elements of the structure need to be initialized.
-		 */
-		priv->dev.ops = &g_audioops;
-		priv->lower = lower;
-		priv->i2c = i2c;
-		priv->i2s = i2s;
 
-		sem_init(&priv->devsem, 0, 1);
-		sq_init(&priv->pendq);
+	if (!priv) {
+		return NULL;
+	}
 
-		/* Software reset.  This puts all ALC5658 registers back in their
-		 * default state.
-		 */
-		alc5658_writereg(priv, ALC5658_SW_RESET, 0);
-		alc5658_dump_registers(&priv->dev, "After reset");
-
-		/* Verify that ALC5658 is present and available on this I2C */
-		regval = alc5658_readreg(priv, ALC5658_SW_RESET);
-		if (regval != 0) {
-			auddbg("ERROR: ALC5658 not found: ID=%04x\n", regval);
-			goto errout_with_dev;
-		}
+	/* Initialize the ALC5658 device structure.  Since we used kmm_zalloc,
+	 * only the non-zero elements of the structure need to be initialized.
+	 */
+	priv->dev.ops = &g_audioops;
+	priv->lower = lower;
+	priv->i2c = i2c;
+	priv->i2s = i2s;
 
```

Comment:
Intialize other elements of priv as well. Like, mute, volume, inout, etc.. feel free to decide on default values and set all these as well.